rocketpool/api/network/dao-proposals.go

```diff
@@ -1,61 +1,184 @@
+		return err
+	})
+
+	// Get the node onchain voting delegate
+	wg.Go(func() error {
+		var err error
+		response.OnchainVotingDelegate, err = network.GetCurrentVotingDelegate(rp, nodeAccount.Address, nil)
+		if err == nil {
+			response.OnchainVotingDelegateFormatted = formatResolvedAddress(c, response.OnchainVotingDelegate)
+		}
+		return err
+	})
+
+	// Get latest block number
+	wg.Go(func() error {
+		_blockNumber, err := ec.BlockNumber(context.Background())
+		if err != nil {
+			return fmt.Errorf("Error getting block number: %w", err)
+		}
+		blockNumber = _blockNumber
+		return nil
+	})
+
+	// Check if Node is registered
+	wg.Go(func() error {
+		var err error
+		response.IsNodeRegistered, err = node.GetNodeExists(rp, nodeAccount.Address, nil)
+		return err
+	})
+
+	// Get active and past votes from Snapshot, but treat errors as non-Fatal
+	if reg != nil {
+		wg.Go(func() error {
+			var err error
+			r := &response.SnapshotResponse
+			if cfg.Smartnode.GetRocketSignerRegistryAddress() != "" {
+				response.SignallingAddress, err = reg.NodeToSigner(&bind.CallOpts{}, nodeAccount.Address)
+				if err != nil {
+					r.Error = err.Error()
+					return nil
+				}
+				blankAddress := common.Address{}
+				if response.SignallingAddress != blankAddress {
+					response.SignallingAddressFormatted = formatResolvedAddress(c, response.SignallingAddress)
+				}
+				votedProposals, err := pdao.GetSnapshotVotedProposals(cfg.Smartnode.GetSnapshotApiDomain(), cfg.Smartnode.GetSnapshotID(), nodeAccount.Address, response.SignallingAddress)
+				if err != nil {
+					r.Error = err.Error()
+					return nil
+				}
+				r.ProposalVotes = votedProposals.Data.Votes
+			}
+			snapshotResponse, err := pdao.GetSnapshotProposals(cfg.Smartnode.GetSnapshotApiDomain(), cfg.Smartnode.GetSnapshotID(), "active")
+			if err != nil {
+				r.Error = err.Error()
+				return nil
+			}
+			r.ActiveSnapshotProposals = snapshotResponse.Data.Proposals
+			return nil
+		})
 	}
 
-	// Get snapshot proposals
-	snapshotResponse, err := node.GetSnapshotProposals(cfg.Smartnode.GetSnapshotApiDomain(), cfg.Smartnode.GetSnapshotID(), "active")
+	// Wait for data
+	if err := wg.Wait(); err != nil {
+		return nil, err
+	}
+
+	// Cast to uint32
+	response.BlockNumber = uint32(blockNumber)
+
+	// Get the proposal artifacts
+	propMgr, err := proposals.NewProposalManager(nil, cfg, rp, bc)
 	if err != nil {
 		return nil, err
 	}
 
-	// Get delegate address
-	idHash := cfg.Smartnode.GetVotingSnapshotID()
-	response.VotingDelegate, err = s.Delegation(nil, nodeAccount.Address, idHash)
+	// Get the delegated voting power if voting is initialized
+	if response.IsVotingInitialized {
+		totalDelegatedVP, _, _, err := propMgr.GetArtifactsForVoting(response.BlockNumber, nodeAccount.Address)
+		if err != nil {
+			return nil, err
+		}
+		response.TotalDelegatedVp = totalDelegatedVP
+	} else {
+		response.TotalDelegatedVp = nil
```

Comment:
these get us into trouble otherwise
```suggestion
		response.TotalDelegatedVp = big.NewInt(0)
```